### PR TITLE
Delete CommonlyUsedGenericInstantiations

### PR DESCRIPTION
**Pull request by stephentoub**
The mechanism is no longer needed, its contents is stale, and it prevents some functionality from being trimmed.

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Internal.cs [hunk 1]
```diff
@@ -1,365 +0,0 @@
...
-using System.Runtime.CompilerServices;
-using System.StubHelpers;
-using System.Threading.Tasks;
-
-#if FEATURE_COMINTEROP
-using System.Runtime.InteropServices.WindowsRuntime;
-#endif // FEATURE_COMINTEROP
-
-[assembly: CLSCompliant(true)]
-[assembly: ComVisible(false)]
```
Comment (MichalStrehovsky) [inline src/System.Private.CoreLib/src/System/Internal.cs]: Should we move the attributes someplace else?

File: src/System.Private.CoreLib/src/System/Internal.cs [hunk 2]
```diff
@@ -1,365 +0,0 @@
...
-
-        // Similar to CommonlyUsedGenericInstantiations but for instantiations of marshaling stubs used
-        // for WinRT redirected interfaces. Note that we do care about reference types here as well because,
-        // say, IList<string> and IList<object> cannot share marshaling stubs.
-        // The methods below "call" most commonly used stub methods on redirected interfaces and take arguments
-        // typed as matching instantiations of mscorlib copies of WinRT interfaces (IIterable<T>, IVector<T>,
-        // IMap<K, V>, ...) which is necessary to generate all required IL stubs.
-
-        [MethodImplAttribute(MethodImplOptions.NoOptimization)]
-        private static void CommonlyUsedWinRTRedirectedInterfaceStubs()
```
Comment (MichalStrehovsky) [inline src/System.Private.CoreLib/src/System/Internal.cs]: This is referenced from ILLinkTrim.xml

File: src/System.Private.CoreLib/src/System/Internal.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Internal.cs]: Ah, missed those.  I'll leave the file and just delete other gunk.

File: src/System.Private.CoreLib/src/System/Internal.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Internal.cs]: Will remove it. Thanks.

**Pull request merged**